rest_api/test/test_rest_api.py

```diff
@@ -66,86 +70,49 @@ def populated_client(client: TestClient) -> TestClient:
             url="/file-upload", files=fi, data={"meta": f'{{"meta_key": "meta_value", "meta_index": "{index}"}}'}
         )
         assert 200 == response.status_code
-    yield client
-    client.post(url="/documents/delete_by_filters", data='{"filters": {}}')
-
-
-def test_get_documents():
-    os.environ["PIPELINE_YAML_PATH"] = str(
-        (Path(__file__).parent / "samples" / "pipeline" / "test_pipeline.yaml").absolute()
-    )
-    os.environ["INDEXING_PIPELINE_NAME"] = "indexing_text_pipeline"
-    client = TestClient(app)
 
-    # Clean up to make sure the docstore is empty
-    client.post(url="/documents/delete_by_filters", data='{"filters": {}}')
+    yield client
 
-    # Upload the files
-    files_to_upload = [
-        {"files": (Path(__file__).parent / "samples" / "docs" / "doc_1.txt").open("rb")},
-        {"files": (Path(__file__).parent / "samples" / "docs" / "doc_2.txt").open("rb")},
-    ]
-    for index, fi in enumerate(files_to_upload):
-        response = client.post(url="/file-upload", files=fi, data={"meta": f'{{"meta_key": "meta_value_get"}}'})
-        assert 200 == response.status_code
 
+def test_get_documents(populated_client: TestClient):
     # Get the documents
-    response = client.post(url="/documents/get_by_filters", data='{"filters": {"meta_key": ["meta_value_get"]}}')
+    response = populated_client.post(url="/documents/get_by_filters", data='{"filters": {"meta_key": ["meta_value"]}}')
     assert 200 == response.status_code
     response_json = response.json()
 
     # Make sure the right docs are found
     assert len(response_json) == 2
```

Comment:
This test is failing at the moment.
```
FAILED rest_api/test/test_rest_api.py::test_get_documents - assert 3 == 2
FAILED rest_api/test/test_rest_api.py::test_delete_documents - assert 3 == 2
```

Reply:
Could it be that one of the PDF files that are uploaded ends up as two Haystack documents, which would be fine?

Reply:
What's interesting is that locally they pass as they are. However I will make them invariant to the document number, so this problem won't show up on the CI :+1:

Reply:
@ZanSara That's a bit worrying. Could explore that further?

Reply:
I agree it's weird, but I-'m not sure I'd call it worrying. 

When I run the tests locally, the two PDFs generate one document each, so I end up with two. On the CI instead I often get a different number of them (like here there were three). On the other hand there are many things that might influence this: for example I run the tests on Python3.9 while the CI is on 3.7. Maybe some dependency in the indexing part influences the splitting. We could investigate it further but I'm not sure it's worth it right now...